Adjust to golangci-lint 2.6.1 changes

tpantelis opened the pull request:
See commits for details.

tpantelis committed:
Use wsl_v5 linter instead of deprecated wsl

...and fix errors, specifically "missing whitespace above this line".

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

skitt commented on .golangci.yml:
```diff
@@ -70,8 +70,11 @@ linters:
         - -QF1008
-    wsl:
-      allow-trailing-comment: true
+    wsl_v5:
```
Commit message...

skitt reviewed (changes requested):
Authorship info is incorrect.

tpantelis committed:
Add new golangci-lint linters

...some commented out if not needed.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Package has more than one godoc

...flagged by the 'godoclint' linter.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
godoc should start with symbol name

...flagged by the 'godoclint' linter.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Replace 'interface{}' with 'any'

...flagged by the 'modernize' linter.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Omitempty has no effect on nested struct fields

...flagged by the 'modernize' linter. Ignore this for CRDs as
   specifying "omitempty" is the norm.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
reflect.TypeOf call can be simplified using TypeFor

...flagged by the 'modernize' linter.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis committed:
Merge branch 'devel' into adj_linters
